### PR TITLE
fix!: remove startup logs eof for streaming

### PR DESCRIPTION
**Pull request by kylecarbs**
We have external utilities like logstream-kube that may send logs after an agent shuts down unexpectedly to report additional information. In a recent change we stopped accepting these logs, which broke these utilities.

In the future we'll rename startup logs to agent logs or something more generalized so this is less confusing in the future.

### CODE REVIEW

File: coderd/workspaceagents.go [hunk 1]
```diff
@@ -580,9 +560,7 @@ func (api *API) workspaceAgentStartupLogs(rw http.ResponseWriter, r *http.Reques
 				continue
 			}
 			if len(logs) == 0 {
-				if state.ReadyAt.Valid {
-					return
-				}
+				// Just keep listening - more logs might come in the future!
```
Comment (mafredri) [inline coderd/workspaceagents.go:563]: We can remove the ready state stuff higher up too.

**Review by mafredri — COMMENTED** [review]
I think this is fine but I can give it a proper look tomorrow.

**Comment by kylecarbs** [comment]
@mafredri this can just wait until Monday, no biggie.

**Review by coadler — APPROVED** [review]

**Review by mafredri — APPROVED** [review]
I just remembered the current log fetching in `cli/cliui.Agent` relies on the websocket closing when streaming startup logs.

I pushed a fix for this and as long as the WebUI doesn't depend on the ws closing behavior, I think we're good.

**Comment by mafredri** [comment]
Marked this as breaking since this will break coder server/clients (e.g. older clients will continue streaming startup logs forever, when blocking startup script is used).

**Pull request merged**